### ISSUE TITLE
Error 500 even with php session folder perms set to 7777

### ISSUE THREAD

**Issue by rt-2** [mgp25/Instagram-API#1562]
Hi,
If you could not close the issue as soon as you answer it, like if your answer was always 100%, which is NOT the case.
I went into my [phpinfo();](http://rt-2.net/info.php) on my server, and the session folder is `/var/lib/php/sessions`, I just `sudo chmod 777 /var/lib/php/sessions`, even restarted apache2 and then I tried:
```
<?php

	error_reporting(-1);
	set_time_limit(0);
	date_default_timezone_set('UTC');

	require '../vendor/autoload.php';
	
  
	try {
		$debug = true;
		//$ig = new \InstagramAPI\Instagram($debug);
	} catch (\Exception $e) {
		echo $e->getMessage()."\n";
	}
	//$i->setUser(IG_USER, IG_PASSWORD);
	//$i->getFollowers(3213205371);
	echo 'works';
?>
```
works, but
```
<?php
	error_reporting(-1);
	set_time_limit(0);
	date_default_timezone_set('UTC');
	require '../vendor/autoload.php';
	try {
		$debug = true;
		$ig = new \InstagramAPI\Instagram($debug);
	} catch (\Exception $e) {
		echo $e->getMessage()."\n";
	}
	echo 'works';
?>
```
still returns error 500 and nothing else, even after doing all you said was "the ONLY possible thing that would make this error"

I don't know how to test, I was not familiar with Composer, but with some playing around I managed to run `composer install` without any errors and it created the wanted 'vendor' directory with the 'autoload.php'.
But everything I try returns a blank error 500....

I see you lack a "requirements" section anywhere,, is it possible that this plguin is NOT compatible with PHP5.6 and only compatible with PHP7??

Thank you,
rt-2

**Comment by rt-2** [comment]
Problem is with composer, followed instructions, composer says everything fine, but not working, will post again if needs.
Problem was NOT my web server, web server =  an application that listen to port 80, please go back to school lol.... saying my web server is at fault for an error 500 is like saying the mall management is at fault because x shops doesn't not have a nice customer service,,, completely ridicule diagnostic sir....

**Comment by SteveJobzniak** [comment]
Stop opening new tickets. Use the old one. This is your last chance.

> is it possible that this plguin is NOT compatible with PHP5.6 and only compatible with PHP7??

I just showed you a log of it running on PHP 5.6. The only requirement is PHP 5.6+.

And I have already told you the error: Your PHP is set to `display_errors 0` which means you will never see the exact error reasons.

Enable error display, and it will show the exact reason. And no

**Comment by rt-2** [comment]
1- make a real 'requirements' section;
2- `Fatal error: Class 'InstagramAPI\Instagram' not found in /home/yanick/www/ammiel/index.php on line 34`
3- Please don't tell me I forgot to run `composer install` because I did and I had like a day of upgrade to do for it on this old serv.
4- This ticket was closed thanks!

**Comment by SteveJobzniak** [comment]
1. We don't need any "requirements" section. Composer checks and installs all requirements for you, and verifies that you are running PHP 5.6 or later. And even if you later move the code to a server without composer, our library's constructor always ensures that you are running PHP 5.6 or later (which is the _only_ requirement). There's no need for any requirements section. It would just say: "Install via composer".
2. See, like I said in the two closed tickets, I told you that enabling error display would show the actual error. It **was** a problem with **your** server: You either hadn't uploaded the "vendor"-folder to the correct location, or perhaps you used the wrong `require __DIR__.'../vendor/autoload.php';` path so that it wasn't able to find the autoload script. Either of those are **your fault** and not a problem with the library, which is why I closed your 3 tickets (after telling you how to enable error reporting so that you would have seen the real error). We could have kept talking in the closed tickets. No need for opening new ones constantly.
3. `composer install` is not the right command. `composer install mgp25/instagram-php` is the right command, as the readme explains. Perhaps that's what went wrong in your case.
4. Alright, despite the attitude I'm glad you got the problems solved at last. ;-)

**Comment by rt-2** [comment]
Yes,
first of all you didn't say to enable php error, you said that my server install was wrong, that there was a problem with the "web server", which is 'apache2' in this case.
By the way the 2 other tickets you close I cannot comment on them.
And yes my mistake is that I tried to understand how composer work and I went to the documentation and I mistakenly too the wrong command to install them. I was not familiar with Composer and I thought it did like supposed to since it used the `composer.json`

**Comment by SteveJobzniak** [comment]
Hi. Normally people can speak in closed tickets, like you are doing right now, here in this closed ticket. If you had done all discussion in the first closed ticket it would have been better and clearer for everyone who reads about this issue in the future. But after you had opened 3 tickets, I decided to actually **lock** the two earlier duplicates, to try to focus all discussion in one place (here). I also renamed all tickets to the same title to try to reduce the damage. ;-)

Anyway, my first reply to you was basically saying that I was sure that something in your web server configuration was wrong. And it was (the PHP config was set to hide all errors). It seems like you were confused by the phrase "web server" and thought I was talking about Apache?

When I said "web server" I wasn't talking about Apache; I meant the whole server machine: https://en.m.wikipedia.org/wiki/Web_server

> The term **can either refer to the entire system**, or specifically to the software that accepts and supervises the HTTP requests.

Anyway, it's great that people can find this ticket if they get error 500.

The answer is to enable error reporting via either php.ini, or via a command like this (if your PHP allows the ini_set command):

```php
<?php

ini_set("display_errors", 1);
```

**Comment by rt-2** [comment]
Not sure why you are getting out of your way NOW to try to tell me everything I already done.
You'd be a nice guy if you'd have posted this at first instead of denying any help, blaming it on "the web server" which you say you refer to the ENTIRE physical machine, which OBVIOUSLY yes the problem is there somewhere lol.
So you basically denied help, then 3 days after the fact give all the help possible to look good. :+1: 
I'm sorry if I opened more issues, it's just that it's the first time that the person doesn't polity ask me if we can close the issue, I guess everyone works differently I just work with different people.
Have a good day.
rt-2

**Comment by rt-2** [comment]
The core of my problem was Composer, so if you want to archive your issues by personal or technical issue, it's your choice......

**Comment by SteveJobzniak** [comment]
Actually, from day one, your ticket was about "only seeing Error 500", and I replied that the reason for that was most likely because your webserver config was wrong. And it was. You had disabled all PHP error output, which is why you only saw "Error 500". Which you would have solved via the 1st search result if you had Googled "php error 500" issue instead of asking here first.

The problem was "Error 500". There was no "Composer Installation/Configuration problem", since you hadn't even executed our composer installation command _at all_. The problem was _literally_ that you didn't have the library at all (you hadn't installed it), and that you had disabled all PHP error output on your server so that you only saw "Error 500" instead of seeing that our library was missing.

Please, do not edit these titles again. I made all of your tickets have the same title so that your obnoxious duplicates would all be grouped with the same name. And the issue you needed help with was "Error 500", and that was what you got help with. After finally enabling error display, you saw that the failure-reason was that you hadn't installed the library at all. So just so we're clear: There is no composer problem. Only a server configuration issue, which then revealed a lack of having installed our library at all via the command we clearly list in our readme:

https://github.com/mgp25/Instagram-API#installation

**Comment by rt-2** [comment]
Honestly you are puting way too much heart into insulting me,,, the reason I say this,,, your 5 paragraphs of text that I would be paid to read all,, well I read the last sentence and I am sure that it resumes well,,,, As I just told you,,, but you ignore 3/4 of what I say every messages, but well lets try again for the 3rd time: my intention was not to 'spam' you,,? I am just extremly unfamiliar with having a ticket closed in the middle of a discution... I heavily use github but for a short amount of time, and the people I work with we have a very different approach.

If you could please keep this to a mature level, I mean the extremly long text you just spawn shows how you are triggered right now but I dont know how old you are but please I dont care what you think please keep yoir emotions to yourself and keep the discution technical....

no need for this tantrum I am sure?

**Comment by SteveJobzniak** [comment]
I'm not "triggered" at all. I'm tired. I'll summarize what you did and why we considered banning you:

- You: "I am getting blank pages with **error 500**! Help!"
- Me: "That 'error 500' has nothing to do with our library. Check if you have disabled error display on your server, since that could cause those blank pages." -closed ticket-
- You: -opens new duplicate ticket- "I am getting blank pages with **error 500**! It's your library's fault! I now have some code that PROVES it!"
- Me: -wastes my time to run your code and shows you the logs proving that your "it's broken!"-code actually works- "See, the library works with your example code. You need to enable error display on your server to see what's _actually_ wrong. (It might be the session-storage, since others have had write-permissions with that before). Here, I've googled the solution for your Error 500 problems (the issue you contacted us about); _this_ is what you need to do to solve the Error 500 issue and actually see the real errors on your server: display_errors 1." -closed ticket-
- You: -new ticket, your 3rd duplicate spammy ticket now (and you apparently still weren't reading my earlier replies telling you to enable error-display)- "I AM STILL GETTING BLANK PAGES WITH ERROR 500 AND NOTHING ELSE! Stop closing my tickets like you know everything!" ... "Oh wait... I FINALLY enabled error display like you told me, and I now see `Cannot find class \InstagramAPI\Instagram`... COMPOSER IS BROKEN! AND I WAITED 3 DAYS TO GET HELP!"
- Me: "No, composer is _not_ broken. It's been less than 24 hours. I helped you with the Error 500, which is what your three duplicate tickets were about. And the error you are now _finally_ seeing means that you haven't even installed our library, which means that you didn't run the installation command clearly listed in our readme." (which you indeed hadn't done)
- You: "How old are you? You are having a tantrum!"

I don't have to be "professional" when someone keeps coming back creating 3 different tickets about "Error 500", and actually _still_ gets help from us with their Error 500 (we _considered_ banning you), and then repeatedly ignores that help and keeps blaming our library (which you hadn't even installed), and Composer (which you simply hadn't executed properly).

I'll give you one more chance to just move on now. If you can act nicely (and search before you ask, and only open 1 ticket about your issue), we'll gladly help you in the future. Good luck!

**Comment by rt-2** [comment]
Again 5 paragraphs, sorry I cannot read that I just got home from helping my mother at her camping for 10h, I'm to tired to read all that, will probably go through those 1300+ words(counted)  from those 2 last messages next week or the next if I have time.
rt-2

**Comment by rt-2** [comment]
P.S. I never said "COMPOSER IS BROKEN! AND I WAITED 3 DAYS TO GET HELP!", first of all if I recall I received response from you the same day so please stop puting words in my mouth I haven't said, its been incredibly difficult to talk to you because you don't seem to understand one word I say, and misinterpret everything I say,,, 
I never said that,, what is your actual problem? you hate me for what, what the fuck,,,
seriously what the fuck, you have some issues man.
I never said "I'VE PROVEN THAT IT'S YOUR LIBRARY'S FAULT!" You are having issues man, stop saying stuff like that... I just said that I didn't know where to further investigate since I wasn't familiar with your API and composer...
As I read through your message right now I see that you are having hard mental issue.
1 I never said I waited 3 days, like you sais I was impressed on how quick you responded. 2 I never said I have code that "proves" that simply including your API makes a server crash, obviously this is widely used so I was the problem, I'm not THAT dumb...
So basically you are saying I said all this horrible stuff but I didn't.... 
Thank you very much.
Sorry for my english,
rt-2

**Comment by SteveJobzniak** [comment]
> **I never said I waited 3 days**

Really? https://github.com/mgp25/Instagram-API/issues/1562#issuecomment-329979521

> You'd be a nice guy if you'd have posted this at first instead of denying any help, blaming it on "the web server" which you say you refer to the ENTIRE physical machine, which OBVIOUSLY yes the problem is there somewhere lol. And it was not the configuration of my web server by the way it was my composer installation... 
So **you basically denied help, then 3 days after the fact give all the help possible** to look good

That's a direct quote. You are saying that we denied helping you for 3 days "by blaming it on your server" (which **was actually the reason for your Error 500 problem**), and that we then suddenly helped you 3 days later. What the fuck? All of your 3 duplicate rage-tickets were **all created** and answered 24 hours ago.

Now you're suddenly forgetting that **_you_ said** we took 3 days to help you, and you're saying that I am mentally insane for saying that you said 3 days. But that's a _direct quote_ from you.

And as for you blaming Composer: There are at least 3 posts above by you where you say that Composer was the problem, and you even tried to change the title of this ticket to "composer installation problem".

As for your immature rage, look:

1st post of this ticket and your 2nd post immediately after it (and there's much more in other tickets and posts but this is enough to prove the point that you are lying):

> still returns error 500 and nothing else, **even after doing all you said** was "the ONLY possible thing that would make this error"
>**Problem is with composer, followed instructions**, composer says everything fine, but not working, will post again if needs.
>**Problem was NOT my web server**, web server =  an application that listen to port 80, **please go back to school lol....** saying my web server is at fault for an error 500 is like saying the mall management is at fault because x shops doesn't not have a nice customer service,,, completely ridicule diagnostic sir....

Wrong. You had never read my posts properly and definitely didn't do what I told you to do, and you didn't even properly understand what I meant about the sessions folder (which is indeed the only thing that could have made this error happen IF YOU HAD INSTALLED OUR LIBRARY, but as it turned out, you hadn't even installed out library at all).

The only problem **you reported to us** was "I am getting blank Error 500 pages!!!". I constantly told you to fix your server configuration to fix your "blank Error 500 pages problem" (THAT'S THE PROBLEM YOU REPORTED TO US, BUT YOU KEEP FORGETTING THAT). You ignored all of my correct advice (about how to enable your server's error display to solve your blank Error 500 pages) and kept blaming our code and composer, until you finally listened and fixed your server config to enable the error message, which revealed that you are an idiot who hadn't even installed our library.

All problems were 100% your fault. You kept blaming the library. You kept spamming and opening new rage-filled and immature tickets. You never listened to what I was saying (telling you to enable error-display on your server to fix the Error 500 problem you wanted help with).

Even after all of that, I kept giving you multiple chances to let it go and even ask us questions in the future if you wanted any help after you got over this bad start. But now I see why you were dumb enough to misunderstand all of my solutions to your problem, and why you were dumb enough to not even have installed our library via the single command we list at the top of our readme, and why you are dumb enough to constantly forget your own words, and why you are dumb and immature enough that you come back with constant insults instead of saying "thanks for not banning me even after all the immature shit I did".

It turns out that you are an absolute idiot who has severe issues with understanding and with rage. I guess I should expect it from someone with a psychotic clown as their profile picture. And you will be getting your well-deserved ban.
